src/lib/import_export/binary/binary_writer.hpp

```diff
@@ -21,6 +21,7 @@ enum class CompressedVectorType : uint8_t;
 class BinaryWriter {
  public:
   static void write(const Table& table, const std::string& filename);
+  friend class StorageManager;
```

Comment:
Fyi, this is needed for the `BinaryWriter::_compressed_vector_type_id` call in `StorageManager::write_dict_segment_to_disk`

Its basically a first start for the removal of the copied files from the binary writer